File: src/test/java/com/cp/csms/authentication/validation/TokenLengthValidatorTest.java
package com.cp.csms.authentication.validation;

import com.cp.csms.common.AuthenticationMessage;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(MockitoExtension.class)
class TokenLengthValidatorTest {

    private static final String REQUEST_ID = "req-123";

    @InjectMocks
    private TokenLengthValidator validator;

    @Test
    void shouldReturnTrueWhenTokenLengthIsMinimum() {
        final String token = "a".repeat(20);
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, token);

        final boolean result = validator.validate(message);

        assertThat(result).isTrue();
    }

    @Test
    void shouldReturnTrueWhenTokenLengthIsMaximum() {
        final String token = "a".repeat(80);
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, token);

        final boolean result = validator.validate(message);

        assertThat(result).isTrue();
    }

    @Test
    void shouldReturnTrueWhenTokenLengthIsWithinRange() {
        final String token = "a".repeat(50);
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, token);

        final boolean result = validator.validate(message);

        assertThat(result).isTrue();
    }

    @Test
    void shouldReturnFalseWhenTokenIsTooShort() {
        final String token = "a".repeat(19);
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, token);

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }

    @Test
    void shouldReturnFalseWhenTokenIsTooLong() {
        final String token = "a".repeat(81);
        final AuthenticationMessage message = new AuthenticationMessage(REQUEST_ID, token);

        final boolean result = validator.validate(message);

        assertThat(result).isFalse();
    }

}

File: src/test/kotlin/com/cp/csms/authentication/validation/TokenLengthValidatorTest.kt
package com.cp.csms.authentication.validation

import com.cp.csms.common.AuthenticationMessage
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.InjectMocks
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class TokenLengthValidatorTest {

    @InjectMocks
    private lateinit var validator: TokenLengthValidator

    @Test
    fun `should return true when token length is minimum`() {
        val token = "a".repeat(20)
        val message = AuthenticationMessage(REQUEST_ID, token)

        val result = validator.validate(message)

        assertThat(result).isTrue()
    }

    @Test
    fun `should return true when token length is maximum`() {
        val token = "a".repeat(80)
        val message = AuthenticationMessage(REQUEST_ID, token)

        val result = validator.validate(message)

        assertThat(result).isTrue()
    }

    @Test
    fun `should return true when token length is within range`() {
        val token = "a".repeat(50)
        val message = AuthenticationMessage(REQUEST_ID, token)

        val result = validator.validate(message)

        assertThat(result).isTrue()
    }

    @Test
    fun `should return false when token is too short`() {
        val token = "a".repeat(19)
        val message = AuthenticationMessage(REQUEST_ID, token)

        val result = validator.validate(message)

        assertThat(result).isFalse()
    }

    @Test
    fun `should return false when token is too long`() {
        val token = "a".repeat(81)
        val message = AuthenticationMessage(REQUEST_ID, token)

        val result = validator.validate(message)

        assertThat(result).isFalse()
    }

    companion object {
        private const val REQUEST_ID = "req-123"
    }
}
